src/main/scala/cognite/spark/v1/DataModelInstancesRelation.scala

```diff
@@ -89,6 +89,52 @@ class DataModelInstanceRelation(config: RelationConfig, modelExternalId: String)
     )
   }
 
+  // scalastyle:off cyclomatic.complexity
+  def getInstanceFilter(sparkFilter: Filter): Seq[DataModelInstanceFilter] =
+    sparkFilter match {
+      case EqualTo(left, right) => {
+        Seq(DMIEqualsFilter(Seq(modelExternalId, left), parseValue(right)))
+      }
+      case In(attribute, values) =>
+        if (Seq(
+            "text[]",
+            "boolean[]",
+            "numeric[]",
+            "float32[]",
+            "float64[]",
+            "int32[]",
+            "int[]",
+            "int64[]",
+            "bigint[]") contains propertyTypes(attribute)._1) {
+          Seq()
+        } else {
+          val setValues = values.filter(_ != null)
+          Seq(DMIInFilter(Seq(modelExternalId, attribute), setValues.map(parseValue)))
+        }
+      case GreaterThanOrEqual(attribute, value) =>
+        Seq(DMIRangeFilter(Seq(modelExternalId, attribute), gte = Some(parseValue(value))))
+      case GreaterThan(attribute, value) =>
+        Seq(DMIRangeFilter(Seq(modelExternalId, attribute), gt = Some(parseValue(value))))
+      case LessThanOrEqual(attribute, value) =>
+        Seq(DMIRangeFilter(Seq(modelExternalId, attribute), lte = Some(parseValue(value))))
+      case LessThan(attribute, value) =>
+        Seq(DMIRangeFilter(Seq(modelExternalId, attribute), lt = Some(parseValue(value))))
+      case And(f1, f2) =>
+        val instancef1 = getInstanceFilter(f1)
+        val instancef2 = getInstanceFilter(f2)
+        Seq(DMIAndFilter(instancef1 ++ instancef2))
+      case Or(f1, f2) =>
+        val instancef1 = getInstanceFilter(f1)
+        val instancef2 = getInstanceFilter(f2)
+        Seq(DMIOrFilter(instancef1 ++ instancef2))
+      case IsNotNull(attribute) =>
+        Seq(DMIExistsFilter(Seq(modelExternalId, attribute)))
+      case Not(f) =>
+        Seq(DMINotFilter(getInstanceFilter(f).head))
```

Comment:
I think that you should probably check the case when getInstanceFilter returns an empty collection (it may be an unsupported filter)

```suggestion
        getInstanceFilter(f).map(DMINotFilter(_))
```

Reply:
I couldn't understand this one :'(